### PR TITLE
8265103: Remove unnecessary inclusion of oopMap.hpp

### PR DESCRIPTION
**Pull request by iklam**
oopMap.hpp is included by about 585 out of about 1000 .o files in HotSpot. In turn, it pulls in other complex headers such as vmreg.hpp, register.hpp and their cpu-dependent headers.

There are only 4 header files that include oopMap.hpp. All of these can be replaced for forward declarations.

According to https://github.com/iklam/tools/tree/main/headers

```
[before]

/jdk/src/hotspot$ whoincludes oopMap.hpp
scanning 585 oopMap.hpp
   2 found 553 method.hpp
   3 found 407 codeBlob.hpp
   4 found 262 debugInfoRec.hpp
   5 found 232 ciEnv.hpp
/jdk/src/hotspot$ whoincludes vmreg.hpp
scanning 597 vmreg.hpp
   2 found 585 oopMap.hpp
   3 found 281 location.hpp
   4 found 118 regmask.hpp
   5 found 109 registerMap.hpp
   ...
/jdk/src/hotspot$ whoincludes register.hpp
scanning 611 register.hpp
   2 found 611 register_x86.hpp
   3 found 597 vmreg.hpp
   4 found 364 assembler.hpp
   5 found 364 assembler_x86.hpp

[after]

/jdk/src/hotspot$ whoincludes oopMap.hpp
scanning 39 oopMap.hpp
   2 found 6 c1_LinearScan.hpp
/jdk/src/hotspot$ whoincludes vmreg.hpp
scanning 333 vmreg.hpp
   2 found 281 location.hpp
   3 found 181 sharedRuntime.hpp
   4 found 118 regmask.hpp
   5 found 109 registerMap.hpp
   ...
/jdk/src/hotspot$ whoincludes register.hpp
scanning 386 register.hpp
   2 found 386 register_x86.hpp
   3 found 364 assembler.hpp
   4 found 364 assembler_x86.hpp
   5 found 333 vmreg.hpp
```

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8265103](https://bugs.openjdk.java.net/browse/JDK-8265103): Remove unnecessary inclusion of oopMap.hpp

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to 80a7f91d6a6041a189561310f8e611f3eb6c0f5a
 * [Stefan Karlsson](https://openjdk.java.net/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to 80a7f91d6a6041a189561310f8e611f3eb6c0f5a

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3446/head:pull/3446` \
`$ git checkout pull/3446`

Update a local copy of the PR: \
`$ git checkout pull/3446` \
`$ git pull https://git.openjdk.java.net/jdk pull/3446/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3446`

View PR using the GUI difftool: \
`$ git pr show -t 3446`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3446.diff">https://git.openjdk.java.net/jdk/pull/3446.diff</a>

</details>

### CODE REVIEW

**Review by dholmes-ora — APPROVED** [review]
Seems okay.

Thanks,
David

File: src/hotspot/share/c1/c1_LinearScan.hpp [hunk 1]
```diff
@@ -31,6 +31,7 @@
 #include "c1/c1_Instruction.hpp"
 #include "c1/c1_LIR.hpp"
 #include "c1/c1_LIRGenerator.hpp"
+#include "compiler/oopMap.hpp"
```
Comment (stefank) [inline src/hotspot/share/c1/c1_LinearScan.hpp:34]: The description states:
> There are only 4 header files that include oopMap.hpp. All of these can be replaced for forward declarations.

But this had to be added. Did you check how many files include c1_LinearScan.hpp?

**Review by stefank — APPROVED** [review]

File: src/hotspot/share/c1/c1_LinearScan.hpp (hunk 1 above)
Comment (iklam) [reply src/hotspot/share/c1/c1_LinearScan.hpp:34]: > The description states:
> 
> > There are only 4 header files that include oopMap.hpp. All of these can be replaced for forward declarations.
> 
> But this had to be added. Did you check how many files include c1_LinearScan.hpp?

c1_LinearScan.hpp is included in only 6 .o files.

Comment (stefank) [reply src/hotspot/share/c1/c1_LinearScan.hpp:34]: Great!

**Comment by iklam** [comment]
Thanks @stefank and @dholmes-ora for the review.
Tested again with latest repo: mach5: tier1, builds-tier2, builds-tier3, builds-tier4 and builds-tier5. Also locally: aarch64, arm, ppc64, s390, x86, and zero.
/integrate

**Pull request closed without merging**